src/main/Catalog.js

```diff
@@ -1,33 +1,43 @@
 import React, { useState, useEffect } from "react";
+import { connect } from "react-redux";
 import InfiniteScroll from "react-infinite-scroll-component";
 import SearchForm from "./SearchForm";
 import "./Catalog.css";
 import beersUrl from "../constants";
+import { setLoad } from "../redux/actions.js";
 
-function Catalog() {
+function Catalog(props) {
   const [beerList, setBeerList] = useState([]);
-  const [beerNumber, setBeerNumber] = useState(9);
+  const {
+    searchText,
+    alcoholVol,
+    bitterness,
+    colorEBC,
+    beerAmount,
+    onSetLoadResult,
+  } = props;
 
   useEffect(() => {
-    let xhttp = new XMLHttpRequest();
-    xhttp.onload = function () {
-      if (this.readyState === 4 && this.status === 200) {
-        setBeerList(JSON.parse(xhttp.responseText));
-      }
-    };
-    xhttp.open("GET", `${beersUrl}?per_page=${beerNumber}`, true);
-    xhttp.send(null);
-  }, [beerNumber]);
+    fetch(
+      `${beersUrl}?per_page=${beerAmount}&abv_gt=${alcoholVol}&ibu_gt=${bitterness}&ebc_gt=${colorEBC}${
+        searchText === "" ? `` : `&beer_name=${searchText}`
+      }`
+    )
+      .then((response) => response.json())
+      .then((products) => setBeerList(products))
+      .catch((err) => alert("Ошибка " + err));
```

Comment:
I think we should be more specific and always in English.
Something like that:

```suggestion
      .catch((err) => alert("Something went wrong: " + err));
```